vm/src/compile.rs

```diff
@@ -30,7 +30,7 @@ pub fn compile(
         },
         Mode::Eval => match parser::parse_statement(source) {
             Ok(statement) => {
-                if let &ast::Statement::Expression { ref expression } = &statement {
+                if let &ast::StatementType::Expression { ref expression } = &statement.statement {
```

Comment:
Having both the variable and the attribute named `statement` (i.e. `statement.statement`) is confusing, but I'm not sure how best to address that (as both Statement and StatementType can meaningfully be called a "statement"). I'd prefer not to have to think "now which type of statement is this?" whenever I'm reading through code.

A couple of ideas:

* Rename the struct member to `statement_type`
* Rename `Statement` to `StatementLocation` (or `ParsedStatement`?) and `StatementType` to `Statement`

Reply:
Thanks! I chose LocatedStatement and rename StatementType back to Statement.